test/set-task.js

```diff
@@ -0,0 +1,21 @@
+var expect = require('expect');
+
+var Undertaker = require('../');
+
+describe('set-task', function() {
+  var taker;
+
+  beforeEach(function(done) {
+    taker = new Undertaker();
+    done();
+  });
+
+  it('should check preconditions', function(done) {
+    expect(function() {
+      taker._setTask('v4task', ['v3task1', 'v3task2']);
+    }).toThrow('Task function must be specified for "v4task"' +
+      ', instead got "v3task1,v3task2"');
```

Comment:
I wonder if we can do something better than hoping the `toString` method of the thing we are logging shows appropriate results.